agent-tooling/src/main/java/io/opentelemetry/auto/typed/base/BaseTypedSpan.java

```diff
@@ -0,0 +1,17 @@
+package io.opentelemetry.auto.typed.base;
+
+import io.opentelemetry.trace.Span;
+
+public abstract class BaseTypedSpan<T extends BaseTypedSpan> extends DelegatingSpan {
+
+  public BaseTypedSpan(Span delegate) {
+    super(delegate);
+  }
+
+  public void end(Throwable throwable) {
+    // add error details to the span.
+    super.end();
+  }
+
+  protected abstract T self();
```

Comment:
This doesn't seem to be used?

Reply:
I think it will still be necessary if we override any methods that return T.